src/demo_parser.rs

```diff
@@ -122,6 +133,53 @@ pub fn parse_directory<'a>(
     )(i)
 }
 
+/// Parse a fallback directory for demo files that were not finalized by a client.
+pub fn parse_fallback_directory<'a>(
+    frames_start: &'a [u8],
+    file_start: &'a [u8],
+    should_parse_netmessage: bool,
+    aux: AuxRefCell,
+) -> Result<'a, Directory> {
+    let file_length = frames_start.len() as i32;
+    let frame_offset = (file_start.len() - frames_start.len()) as i32;
+
+    let mut frames: Vec<Frame> = vec![];
+    let mut frames_start = &frames_start[..];
+
+    loop {
+        match parse_frame(frames_start, should_parse_netmessage, aux.clone()) {
+            Ok((end_current_frame, frame)) => {
+                frames_start = end_current_frame;
+                frames.push(frame);
+
+                if end_current_frame.is_empty() {
+                    break;
+                }
+            }
+            Err(_) => break,
+        }
+    }
+
+    let fallback_entry = DirectoryEntry {
+        type_: FALLBACK_DIRECTORY_ENTRY_TYPE,
+        description: vec![],
```

Comment:
This looks questionable without any context. `description` is 64 bytes long and my writer doesn't pad description. So reasonably it will fail the parser. But in the parser, you have to ignore reading the directory entry, And then in the writer, you have to ignore writing the directory entry. If you can add some comments here mentioning some of that, it would be helpful when looking back.